components/net/image_cache.rs

```diff
@@ -570,6 +570,25 @@ impl ImageCache for ImageCacheImpl {
         cache_result
     }
 
+    fn decode(
+        &self,
+        url: ServoUrl,
+        origin: ImmutableOrigin,
+        cors_setting: Option<CorsSettings>,
+    ) -> Option<Image> {
+        let mut store = self.store.lock().unwrap();
+
```

Comment:
Thanks for review @gterzian, what confused me is where to actually decode the img, I added this decode function to decode the img, but if I resolve the promise in `handle_loaded_image` I don't need this decode function because image in `handle_loaded_image` is already decoded, right?

Reply:
Yes you are correct.